internal/setting/service.go

```diff
@@ -0,0 +1,116 @@
+package setting
+
+import (
+	"context"
+	"github.com/NYCU-SDC/summer/pkg/database"
+	"github.com/NYCU-SDC/summer/pkg/log"
+	"github.com/google/uuid"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+type Service struct {
+	logger *zap.Logger
+	tracer trace.Tracer
+	query  *Queries
+}
+
+func NewService(logger *zap.Logger, db DBTX) *Service {
+	return &Service{
+		logger: logger,
+		tracer: otel.Tracer("setting/service"),
+		query:  New(db),
+	}
+}
+
+func (s *Service) GetSettingByUserId(ctx context.Context, userId uuid.UUID) (Setting, error) {
+	traceCtx, span := s.tracer.Start(ctx, "GetSettingByUserId")
+	defer span.End()
+	logger := logutil.WithContext(traceCtx, s.logger)
+
+	setting, err := s.query.GetSetting(ctx, userId)
+	if err != nil {
+		err = databaseutil.WrapDBErrorWithKeyValue(err, "settings", "id", userId.String(), logger, "get setting by user id")
+		span.RecordError(err)
+		return Setting{}, err
+	}
+
+	return setting, nil
+}
+
+func (s *Service) UpdateSetting(ctx context.Context, userId uuid.UUID, setting Setting) (Setting, error) {
+	traceCtx, span := s.tracer.Start(ctx, "UpdateSetting")
+	defer span.End()
+	logger := logutil.WithContext(traceCtx, s.logger)
+
+	updatedSetting, err := s.query.UpdateSetting(ctx, UpdateSettingParams(setting))
+
```

Comment:
In other parts of this file, there are no blank lines around if err != nil, so it might be better to remove this blank line for consistency and better readability.